### PR TITLE
HCAP-1464 - add scripts for generating random test data

### PR DESCRIPTION
**Pull request by coryatwater-FW**
- created 2 scripts which **_overwrite_** a few existing test-data files: `participants.csv`, `cohort_participants.csv`, `participants_status.csv`, `participant_post_hire_status.csv`, `return_of_service_status.csv`
- purpose is to capture the entire flow of statuses for participants

 ### randomize-participants

- generates 500 (default) randomized participants for Fraser(default) preferred location
- overwrites `participants.csv`

### randomize-participant-status

- generates data for the following .csv files: `cohort_participants.csv`, `participants_status.csv`, `participant_post_hire_status.csv`, `return_of_service_status.csv`
- runs through 400 loops randomly picking 1 of 8 statuses(partic_statuses) and adds any statuses required to meet chosen statuses required history

### CODE REVIEW

File: server/scripts/randomize-participants-status.ts [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+const returnOfServiceStatuses = ['assigned-same-site', 'assigned-new-site'];
+const positionType = ['casual', 'permanent'];
+const employmentType = ['part-time', 'full-time'];
+
+const randomize = (n: number, reject?: boolean) => {
+  let value = 0;
+  // prevent duplicate participant ids from being used
+  // Sonarcloud complains about Math.random, ignoring as it's just being used for simple random numbers
+  if (reject) {
+    value = Math.floor(Math.random() * n); //NOSONAR
```
Comment (sunghwan-park-fw) [inline server/scripts/randomize-participants-status.ts:61]: How about using [lodash.random](https://lodash.com/docs/4.17.15#random)?

File: server/scripts/randomize-participants-status.ts [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+  await convertToCsv(cId, cohortPartArray, 'cohort_participants.csv');
+  await convertToCsv(pphId, partPostHireArray, 'participant_post_hire_status.csv');
+  await convertToCsv(rosId, returnOfServiceArray, 'return_of_service_status.csv');
+};
+
+// loop through a status and add any related pre-statuses
+const generateFullStatusEntry = (status: string) => {
+  if (status === 'engaged') {
+    isCurrent = true;
+    partStatusArray.push(generateProspectingEntry());
```
Comment (sunghwan-park-fw) [inline server/scripts/randomize-participants-status.ts:106]: how about passing `current` as a param to `generateProspectingEntry?`
We'd better avoid updating or referencing global variables as much as possible.

File: server/scripts/randomize-participants-status.ts [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+  }
+  return Math.floor(Math.random() * n); //NOSONAR
+};
+
+// create a random date based off the previous statuses date
+const randomizeDate = (d?: string) => {
+  const start = dayjs(d || START_DATE);
+  const daysToAdd = randomize(365);
+  const addedDate = start.add(daysToAdd, 'day').format('YYYY/MM/DD');
+  incrementedDate = addedDate;
```
Comment (sunghwan-park-fw) [inline server/scripts/randomize-participants-status.ts:77]: It updates a global variable and returns a local. This pattern is hard to read.

Comment (coryatwater-FW) [reply server/scripts/randomize-participants-status.ts:77]: hmm I can agree with that, I can add a comment to further clarify, but it's purpose is to maintain a consistent flow of dates that are moving forward that all the statuses have access too.  Open to other ideas of course!

File: server/scripts/randomize-participants-status.ts (hunk 2 above)
Comment (coryatwater-FW) [reply server/scripts/randomize-participants-status.ts:106]: Yah, better idea!

**Review by fw-noel — APPROVED** [review]

**Pull request merged**